nixpkgs_review/nix.py

```diff
@@ -38,6 +50,61 @@ def was_build(self) -> bool:
     def is_test(self) -> bool:
         return self.name.startswith("nixosTests")
 
+    def log(self, tail: int = -1, strip_colors: bool = False) -> Optional[str]:
+        def get_log(path: Optional[str]) -> Optional[str]:
+            if path is None:
+                return None
+            system = subprocess.run(
+                ["nix", "--experimental-features", "nix-command", "log", path],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                text=True,
+                errors="backslashreplace",
+            )
+            stdout = system.stdout
+            if tail > 0 and len(stdout) > tail:
+                stdout = "This file has been truncated\n" + stdout[-tail:]
+
+            if strip_colors:
+                return strip_ansi_colors(stdout)
+            return stdout
+
+        if self.drv_path is None:
+            return None
+
+        value = get_log(self.drv_path) or get_log(self.path) or ""
+        if self.build_err_msg is not None:
+            value = "\n".join([value, self.build_err_msg])
+
+        return value
+
+    def log_path(self) -> Optional[str]:
+        if self.drv_path is None:
+            return None
+        base = os.path.basename(self.drv_path)
+        prefix = "/nix/var/log/nix/drvs/"
```

Comment:
This might now always be true.

Reply:
I don't actually know how to determine this location robustly. It could be in a different location if you're running with a sufficiently esoteric `/etc/nix/nix.conf`, and it's not even guaranteed that nix's config file with be in `/etc/nix/nix/conf`.

Reply:
Marked with a "TODO" item

Reply:
Can we use ``_store_dir()`` here?